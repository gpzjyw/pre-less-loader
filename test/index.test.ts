/**
 * @jest-environment node
 */
import path from 'path';
import vm from 'vm';
import { IFs } from 'memfs';
import compile from './util/compile';

test('transform less', async () => {
  const { stats, compiler } = await compile();

  const name = stats.toJson().assets[0].name; 
  const code = (compiler.outputFileSystem as IFs).readFileSync(
    path.join(stats.compilation.outputOptions.path, name)
  ).toString();

  const result = vm.runInNewContext(
    `${code};module.exports = customLoaderExport;`,
    {
      module: {},
    }
  );

  expect(result.default.css).toMatchSnapshot('css');
});
