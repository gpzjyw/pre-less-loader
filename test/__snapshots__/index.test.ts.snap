// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform less: css 1`] = `
"
      /** autoLessVariableExport **/
// 以下存放业务相关的变量
@header-height        : 50px;

@nav-head-icon-size   : 14px;
@nav-side-icon-size   : 18px;
@page-icon-size       : 20px;

// 导航相关
@nav-bg               : #374259;
@nav-side-bg          : #1F2633;
      :export {
        headerHeight: @header-height;
navHeadIconSize: @nav-head-icon-size;
navSideIconSize: @nav-side-icon-size;
pageIconSize: @page-icon-size;
navBg: @nav-bg;
navSideBg: @nav-side-bg;

      };
    "
`;
